lib/service/service.go

```diff
@@ -4426,7 +4433,10 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 			Log:                      log,
 			IngressReporter:          ingressReporter,
 			KubernetesServersWatcher: kubeServerWatcher,
-			PROXYProtocolMode:        cfg.Proxy.PROXYProtocolMode,
+			PROXYProtocolMode:        proxyProtocol,
+
+			// Used for tests to force checking of required PROXY lines for all connections.
+			MultiplexerIgnoreSelfConnections: cfg.MultiplexerIgnoreSelfConnections,
```

Comment:
can we live without exposing this setting to the outside world? 
Adding more config options will cause more troubles than what they fix 
i.e. having optional arguments to change the multiplexer config.